e.u.c.f.b.AnalysisException after upgrade of SB from 4.0.4 to 4.0.5 in maven project

```
[INFO] --- spotbugs-maven-plugin:4.0.4:spotbugs (default-cli) @ api ---
[INFO] Fork Value is true
     [java] Exception in thread "main" edu.umd.cs.findbugs.ba.AnalysisException: Exeption was thrown during analysis
     [java] 	at edu.umd.cs.findbugs.FindBugs2.lambda$analyzeApplication$3(FindBugs2.java:1125)
     [java] 	at java.util.ArrayList.forEach(ArrayList.java:1257)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1118)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:309)
     [java] 	at edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:395)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1231)
     [java] Caused by: java.util.concurrent.ExecutionException: java.lang.AssertionError: frame_type field doesn't exist
     [java] 	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
     [java] 	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.lambda$analyzeApplication$3(FindBugs2.java:1120)
     [java] 	... 5 more
     [java] Caused by: java.lang.AssertionError: frame_type field doesn't exist
     [java] 	at edu.umd.cs.findbugs.StackMapAnalyzer.lambda$static$0(StackMapAnalyzer.java:150)
     [java] 	at java.security.AccessController.doPrivileged(Native Method)
     [java] 	at edu.umd.cs.findbugs.StackMapAnalyzer.<clinit>(StackMapAnalyzer.java:142)
     [java] 	at edu.umd.cs.findbugs.StackMapAnalyzer$StackMapAnalysisFactory.analyze(StackMapAnalyzer.java:66)
     [java] 	at edu.umd.cs.findbugs.StackMapAnalyzer$StackMapAnalysisFactory.analyze(StackMapAnalyzer.java:58)
     [java] 	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.analyzeMethod(AnalysisCache.java:368)
     [java] 	at edu.umd.cs.findbugs.classfile.impl.AnalysisCache.getMethodAnalysis(AnalysisCache.java:321)
     [java] 	at edu.umd.cs.findbugs.OpcodeStack.getJumpInfoFromStackMap(OpcodeStack.java:3075)
     [java] 	at edu.umd.cs.findbugs.OpcodeStack.resetForMethodEntry(OpcodeStack.java:3030)
     [java] 	at com.mebigfatguy.fbcontrib.collect.CollectStatistics.visitCode(CollectStatistics.java:140)
     [java] 	at org.apache.bcel.classfile.Code.accept(Code.java:131)
     [java] 	at edu.umd.cs.findbugs.visitclass.PreorderVisitor.doVisitMethod(PreorderVisitor.java:315)
     [java] 	at edu.umd.cs.findbugs.visitclass.PreorderVisitor.visitJavaClass(PreorderVisitor.java:397)
     [java] 	at org.apache.bcel.classfile.JavaClass.accept(JavaClass.java:213)
     [java] 	at edu.umd.cs.findbugs.BytecodeScanningDetector.visitClassContext(BytecodeScanningDetector.java:38)
     [java] 	at com.mebigfatguy.fbcontrib.collect.CollectStatistics.visitClassContext(CollectStatistics.java:108)
     [java] 	at edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:76)
     [java] 	at edu.umd.cs.findbugs.FindBugs2.lambda$null$1(FindBugs2.java:1108)
     [java] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
     [java] 	at edu.umd.cs.findbugs.CurrentThreadExecutorService.execute(CurrentThreadExecutorService.java:86)
     [java] 	at java.util.concurrent.AbstractExecutorService.invokeAll(AbstractExecutorService.java:238)
```
Project uses
```xml
                            <plugin>
                                <groupId>com.mebigfatguy.sb-contrib</groupId>
                                <artifactId>sb-contrib</artifactId>
                                <version>7.4.7</version>
                            </plugin>
```
as I see
```
     [java] 	at com.mebigfatguy.fbcontrib.collect.CollectStatistics.visitCode(CollectStatistics.java:140)
```
in above log, but it was working fine with 4.0.4 (ie. spotbugs upgrade is the only changed piece that triggers `AnalysisException`).
Given https://github.com/spotbugs/spotbugs/blob/4.0.5/CHANGELOG.md and claim to follow semantic versioning I find this exception a bit surprising.
I see there was
```
-  api 'org.apache.bcel:bcel:6.4.1'
+  api 'org.apache.bcel:bcel:6.5.0'
```
between 4.0.4 and 4.0.5, and bcel also participates in trace.

No separate reproducer at the moment, but I can work on one if the cause isn't obvious and one is needed.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ This is the changelog for SpotBugs. This follows [Keep a Changelog v1.0.0](http:
 Currently the versioning policy of this project follows [Semantic Versioning v2.0.0](http://semver.org/spec/v2.0.0.html).
 
 ## Unreleased - 2020-??-??
+### Fixed
+* Use method call instead of reflection to get BCEL frame type ([#1176](https://github.com/spotbugs/spotbugs/issues/1176))
 
 ## 4.0.5 - 2020-06-20
 ### Fixed

--- a/spotbugs/src/main/java/edu/umd/cs/findbugs/StackMapAnalyzer.java
+++ b/spotbugs/src/main/java/edu/umd/cs/findbugs/StackMapAnalyzer.java
@@ -19,9 +19,6 @@
 
 package edu.umd.cs.findbugs;
 
-import java.lang.reflect.Field;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
 import java.util.ArrayList;
 import java.util.BitSet;
 import java.util.HashMap;
@@ -43,7 +40,6 @@ import org.apache.bcel.generic.Type;
 
 import edu.umd.cs.findbugs.OpcodeStack.Item;
 import edu.umd.cs.findbugs.OpcodeStack.JumpInfo;
-import edu.umd.cs.findbugs.ba.AnalysisContext;
 import edu.umd.cs.findbugs.classfile.CheckedAnalysisException;
 import edu.umd.cs.findbugs.classfile.IAnalysisCache;
 import edu.umd.cs.findbugs.classfile.MethodDescriptor;
@@ -135,52 +131,7 @@ public class StackMapAnalyzer {
         return locals;
     }
 
-    static final @CheckForNull Field frame_type_field;
-    static {
-        Field f;
-        try {
-            f = AccessController.doPrivileged((PrivilegedAction<Field>) () -> {
-                Class<StackMapEntry> c = StackMapEntry.class;
-                Field result;
-                try {
-                    result = c.getDeclaredField("frame_type");
-                    result.setAccessible(true);
-                    return result;
-                } catch (NoSuchFieldException e1) {
-                    throw new AssertionError("frame_type field doesn't exist");
-                } catch (SecurityException e2) {
-                    return null;
-                }
-
-            });
-        } catch (Exception e) {
-            AnalysisContext.logError("Unable to create frame_type accessor", e);
-            f = null;
-        }
-        if (DEBUG) {
-            System.out.println("Frame type field is null:" + (f == null));
-        }
-        frame_type_field = f;
-    }
-
-    static int getFrameType(StackMapEntry e) {
-        if (frame_type_field == null) {
-            return -1;
-        }
-        try {
-            return (Integer) frame_type_field.get(e);
-        } catch (IllegalArgumentException e1) {
-            return -1;
-        } catch (IllegalAccessException e1) {
-            return -1;
-        }
-    }
-
     static private @CheckForNull JumpInfoFromStackMap getFromStackMap(IAnalysisCache analysisCache, MethodDescriptor descriptor) {
-        if (frame_type_field == null) {
-            return null;
-        }
-
         Method method;
         try {
             method = analysisCache.getMethodAnalysis(Method.class, descriptor);
@@ -211,7 +162,7 @@ public class StackMapAnalyzer {
         int pc = 0;
         for (StackMapEntry e : stackMapTable.getStackMap()) {
             pc += e.getByteCodeOffset();
-            int rawFrameType = getFrameType(e);
+            int rawFrameType = e.getFrameType();
             StackFrameType stackFrameType = StackFrameType.get(rawFrameType);
             switch (stackFrameType) {
             case SAME_FRAME:

